### ISSUE TITLE
"Task is stuck" : doesn't finish or perform post-download actions

### ISSUE THREAD

**Issue by fire-eggs** [fffonion/xeHentai#70]
I got a "task is stuck" situation at one point:
```
[Nov 10 00:55:24] task #807f067e is stuck, there may be some bugs in xeHentai, or the connection is too slow
```
Hardly a "fatal" situation, but as I had to terminate the task, all "post-download" activity (e.g. renaming to the original filename) didn't happen.

A better error recovery here would be great. Suggestions for how I might fix it myself also appreciated.

My thanks for a useful script!!!

**Comment by fffonion** [comment]
Hi @fire-eggs, thanks for the report. This issue is known when connection to image cache is slow or blocked, but I don't have enough bandwidth to investigate. Help on this will be welcomed!

As a temporary solution, I would recommand to use the [webui](https://github.com/fffonion/xeHentai-webui) to pause the task and start it again.

**Comment by fire-eggs** [comment]
>I don't have enough bandwidth to investigate
I understand and sympathize. I'll take a look when I can, and thank you for the webui pointer as well!

**Comment by fffonion** [comment]
Hi @fire-eggs , this issue is finally identified and fixed in version 2.022. Please download from releases.

Closing this issue and feel free to open a new one if you still face the same issue!
